src/Umbraco.Core/IO/IOHelper.cs

```diff
@@ -13,7 +13,9 @@
 namespace Umbraco.Core.IO
 {
 	public static class IOHelper
-    {
+	{
+	    public static bool IAmUnitTestingSoNeverUseHttpContextEver = false;

```

Comment:
Yep, injected will be way to go. Wish there was HostingEnvironment.Set(Mode.UnitTest, path).